sklearn/utils/_estimator_html_repr.py

```diff
@@ -78,24 +80,81 @@ def _write_label_html(
     outer_class="sk-label-container",
     inner_class="sk-label",
     checked=False,
+    doc_link="",
+    is_fitted_css_class="",
+    is_fitted_icon="",
 ):
-    """Write labeled html with or without a dropdown with named details"""
-    out.write(f'<div class="{outer_class}"><div class="{inner_class} sk-toggleable">')
+    """Write labeled html with or without a dropdown with named details.
+
+    Parameters
+    ----------
+    out : file-like object
+        The file to write the HTML representation to.
+    name : str
+        The label for the estimator. It corresponds either to the estimator class name
+        for simple estimator or in the case of `Pipeline` and `ColumnTransformer`, it
+        corresponds to the name of the step.
+    name_details : str
+        The details to show as content in the dropdown part of the toggleable label.
+        It can contain information as non-default parameters or column information for
+        `ColumnTransformer`.
+    outer_class : {"sk-label-container", "sk-item"}, default="sk-label-container"
+        The CSS class for the outer container.
+    inner_class : {"sk-label", "sk-estimator"}, default="sk-label"
+        The CSS class for the inner container.
+    checked : bool, default=False
+        Whether the dropdown is checked or not. With a single estimator, we intend to
+        unfold the content.
```

Comment:
Should this be called `folded`? I'm not sure I understand what a "(un)checked dropdown" is. I assume this comes from HTML where a dropdown/foldable content is implemented as something that can be "checked" (like a checkbox)?

But, we didn't touch this parameter name in this PR, so maybe we should just stick with what we have.

Maybe say "Whether the dropdown is folded or not."?

Reply:
> Should this be called folded? I'm not sure I understand what a "(un)checked dropdown" is. I assume this comes from HTML where a dropdown/foldable content is implemented as something that can be "checked" (like a checkbox)?

Yes this is linked to the folding. I can rename the description. I am not sure to change the internal thought since this is not related to this PR. I would prefer to have it in another PR (I assume that we have specific tests as well.).

Reply:
Changing the description is fine with me